custom_components/ha-google-home/api.py

```diff
@@ -52,62 +54,88 @@ def get_android_id(self):
         """Generate random android_id"""
         return self._client.get_android_id()
 
+    @staticmethod
+    def format_offline_devices_to_human_string(device_list):
+        """Removes unwanted char's from string"""
+        device_list = str(device_list)
+        char = "[']"
+        for c in char:
+            device_list = device_list.replace(c, "")
+
+        return device_list
+
     @staticmethod
     def create_url(ip, port, api_endpoint):
-        """Creates url to endpoint.
+        """First it checks if a valid IP have been provided and then it creates a url to endpoint.
         Note: port argument is unused because all request must be done to 8443"""
-        url = "https://{ip}:{port}/{endpoint}".format(
-            ip=ip, port=str(PORT), endpoint=api_endpoint
-        )
-        return url
+        if re.search(IP_CHECK_REGEX, ip):
```

Comment:
Let's better use inbuild `glocaltokens` check that is more robust. Check it [here](https://github.com/leikoilja/glocaltokens/blob/master/glocaltokens/utils/network.py)

`from glocaltokens.utils.network import is_valid_ipv4_address`
It will return True/False

Reply:
okay